assembl/static2/js/app/components/common/avatar.jsx

```diff
@@ -24,10 +25,17 @@ class ProfileIcon extends React.Component {
       next: location
     });
   }
+  redirectToJoin() {
+    const slug = { slug: getDiscussionSlug() };
+    browserHistory.push(get('join', slug));
+  }
```

Comment:
I think you need to push to customBrowserHistory created in index.jsx
You should export the history variable from index.jsx and use it here.